src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -175,74 +176,65 @@ private bool CanUseVar(ITypeSymbol typeSymbol)
             NameSyntax newLocalName,
             LocalDeclarationStatementSyntax declarationStatement,
             bool allOccurrences,
+            bool createReturnStatement,
             CancellationToken cancellationToken)
         {
             var oldBody = arrowExpression;
             var oldParentingNode = oldBody.Parent;
             var leadingTrivia = oldBody.GetLeadingTrivia()
                                        .AddRange(oldBody.ArrowToken.TrailingTrivia);
 
-            var newStatement = Rewrite(document, expression, newLocalName, document, oldBody.Expression, allOccurrences, cancellationToken);
-            var newBody = SyntaxFactory.Block(declarationStatement, SyntaxFactory.ReturnStatement(newStatement))
+            var newExpression = Rewrite(document, expression, newLocalName, document, oldBody.Expression, allOccurrences, cancellationToken);
+
+            var convertedStatement = createReturnStatement
+                ? SyntaxFactory.ReturnStatement(newExpression)
+                : SyntaxFactory.ExpressionStatement(newExpression) as StatementSyntax;
```

Comment:
This should be a cast, since we necessarily have a `StatementSyntax`

Reply:
Yes, but we also know we necessarily have a `StatementSyntax` at compile time. The "cast" only exists to appease the type system. I'll change it, but in reality, neither a cast or 'as' is great - if there was another type of cast that only allowed implicit conversions (as if I just assigned to a variable with this type), that would be the most appropriate.

Reply:
Yes, both a cast and an `as` appease the compiler. But a cast is unambiguous to the human reader.

Reply:
👍 A cast is how we handle this in the rest of hte IDE codebase.  I would def appreciate if you could stick with that informal convention.

Reply:
fixed in c1b12c90a788d95048193d2e2eb0047b3a632b75